src/coreclr/jit/fgbasic.cpp

```diff
@@ -939,6 +949,16 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
         compInlineResult->Note(InlineObservation::CALLEE_BEGIN_OPCODE_SCAN);
     }
 
+    if (preciseScan)
+    {
+        CORINFO_SIG_INFO sig;
+        eeGetMethodSig(info.compMethodHnd, &sig);
+        if (sig.retType == CORINFO_TYPE_CLASS)
+        {
+            exactSigRet = info.compCompHnd->isExactType(sig.retTypeClass);
+        }
```

Comment:
We avoid calling VM apis during inlining - this potentially can be slow/trigger type load events for every inline attempt (and we do many of them)

Reply:
> this potentially can be slow/trigger type load events for every inline attempt (and we do many of them)

We are already resolving method tokens while scanning the inlinee, so I don't see this to be a severe problem.

Reply:
> > this potentially can be slow/trigger type load events for every inline attempt (and we do many of them)
> 
> We are already resolving method tokens for calls while scanning the inlinee, which loads all the type dependencies, so I don't see this to be a problem.

Yes and the initial impl of that caused a massive TP regression as far as I remember, so it's not an excuse to continue calling other VM stuff, if we want to land this, we need to measure/make sure it's cheap